src/pages/waypoint/docs/[...page].tsx

```diff
@@ -0,0 +1,59 @@
+import {
+  GetStaticPathsContext,
+  GetStaticPathsResult,
+  GetStaticPropsContext,
+} from 'next'
+import waypointData from 'data/waypoint.json'
+import { ProductData } from 'types/products'
+import { getStaticGenerationFunctions } from 'layouts/sidebar-sidecar/server'
+import DocsView from 'views/docs-view'
+
+const basePath = 'docs'
+const baseName = 'Docs'
+const product = waypointData as ProductData
+
+const {
+  getStaticPaths: generatedGetStaticPaths,
+  getStaticProps: generatedGetStaticProps,
+} = getStaticGenerationFunctions({
+  product,
+  basePath,
+  baseName,
+})
+
+/**
+ * Wrapper for `generatedGetStaticProps` that first checks if the current page
+ * is `/waypoint/docs/index`. If that is the current page, then the user will be
+ * sent to the 404 page. Otherwise, `generatedGetStaticProps` will be invoked as
+ * usual.
+ */
+export async function getStaticProps(context: GetStaticPropsContext) {
+  if (context.params.page[0] === 'index') {
```

Comment:
⛏️ Nitpick-y thought:

TLDR; I think you could drop this extra layer, and export `getStaticProps` unmodified.

IMO there's no need to include this specific case in the scope of the custom docs landing work. I believe the underlying issue here needs to be fixed instead, as detailed in 🎟️ [[docs-loader] /docs/some-category/index should 404, but does not](https://app.asana.com/0/1202097197789424/1202289240521898/f). I _think_ the only way anyone would get to this working-but-it-shouldn't URL is by directly typing it in; and we have the same problem on live sites (eg see https://www.waypointproject.io/docs/index, https://www.waypointproject.io/docs/intro/index, https://www.waypointproject.io/docs/upgrading/index, etc etc.)

That being said, I suppose there's no harm in including this - though I think we'll want to remember to remove it once we do resolve the above task, so would personally lean towards omitting it (to have to remember about it later! ☺️ ).

Reply:
This sounds totally reasonable to me! Especially considering that it's an existing issue. I'll remove this in my next PR. 😊